lib/adal-aad-node/src/errors/errors.ts

```diff
@@ -0,0 +1,29 @@
+import { ErrorLookup, Error1Context } from "../models/error";
+
+type ErrorMapping = {
+    [errorCode: number]: string;
+};
+
+const simpleErrorMapping: ErrorMapping = {
+    2: 'Something failed with the authentication, or your tokens have been deleted from the system. Please try adding your account to Azure Data Studio again',
+    3: 'Token retrival failed with an error. Open developer tools to view the error',
+    4: 'No access token returned from Microsoft OAuth',
+    5: 'The user had no unique identifier within AAD',
+    6: 'Error retrieving tenant information',
+    7: 'Error when getting your account from the cache',
+    8: 'Error when parsing your account from the cache',
+    9: 'Error when adding your account to the cache',
```

Comment:
Would suggest having an enum that maps to these numeric values so you can use those in the code here instead of hardcoding numbers

Reply:
Isn't using enums that are can randomly change usually considered a code_smell? I was thinking maybe just using string based keys.

Reply:
What do you mean randomly change? I wouldn't expect the number values to actually change...or is that not the case?

And to be clear - I'm suggesting making an enum for the numbers themselves.

```
enum ErrorCodes {
  MyError = 1
  MyOtherError = 2
  SomeOtherReallyBadError = 3
...
}
```

and then you'd keep this mapping and just use those enum values instead of random numbers all over the place.

Reply:
I guess this is more of an issue in languages that can actually iterate over enum values, but this was a major issue with for example Bukkit using a Material enum, where every update introduced new enum values and broke a lot of other people using the Bukkit library - so it kinda stuck with me to not use enums for values that can change in future updates since it can break API contracts.

Either way I do like your suggestion and will prob switch to it.